applications/ConvectionDiffusionApplication/custom_response_functions/local_temperature_average_response_function.h

```diff
@@ -160,8 +160,28 @@ class LocalTemperatureAverageResponseFunction: public AdjointResponseFunction
     double CalculateValue(ModelPart& rModelPart) override
     {
         KRATOS_TRY;
-        KRATOS_ERROR
-            << "PointTemperature::CalculateValue(ModelPart& rModelPart) is not implemented!!!\n";
+	
+        const ModelPart& r_target_model_part = GetTargetModelPart(rModelPart, mTargetModelPartName);	
+
+        const Communicator& r_communicator = r_target_model_part.GetCommunicator();
+        const int number_of_nodes = r_communicator.LocalMesh().NumberOfNodes();
+        
+	double domain_aggregated_temperature = 0.0;
+
+#pragma omp parallel for reduction(+ : domain_aggregated_temperature)
+        for (int i = 0; i < number_of_nodes; ++i)
+        {
+            const ModelPart::NodeType& r_node =
+                *(r_communicator.LocalMesh().NodesBegin() + i);
+            domain_aggregated_temperature += r_node.FastGetSolutionStepValue(TEMPERATURE);
```

Comment:
@sunethwarna I thought the variable utils have a function for this, can you check pls?

Reply:
yes there is.... :) thanks :)

Reply:
I get a compiler error saying that there is no ‘SumHistoricalNodeScalarVariable’

Reply:
I compiled it, and it works fine in my pc :/... thats strange, VariableUtils has the method implemented. @philbucher ?

Reply:
CI is also passing, we have ConvDiff app in CI ryt?

Reply:
hm strange. in variable_utils.h on master it does not exist.

ConvDiff is not in the CI.

Reply:
@armingeiser is right. The `SumHistoricalNodeScalarVariable` was removed and replaced by `SumHistoricalVariable`.